tests/st/test_zero.py

```diff
@@ -0,0 +1,80 @@
+import logging
+
+import numpy as np
+
+import mindspore as ms
+from mindspore import nn
+from mindspore.communication import get_group_size, get_rank, init
+from mindspore.communication.management import GlobalComm
+
+from mindone.trainers.zero import prepare_train_network
+from mindone.utils.logger import set_logger
+
+_logger = logging.getLogger(__name__)
+
+
+def init_env(mode, distribute, save_graph=True):
+    ms.set_seed(1)
+    ms.set_context(mode=mode)
+    if save_graph:
+        ms.set_context(save_graphs=True, save_graphs_path="ms_ir")
+    if distribute:
+        init()
+        group_size = get_group_size()
+        rank_id = get_rank()
+        print(f"rank_id: {rank_id}, group_size: {group_size}")
+        ms.reset_auto_parallel_context()
+        ms.set_auto_parallel_context(
+            parallel_mode=ms.ParallelMode.DATA_PARALLEL,
```

Comment:
How is the difference between it ParallelMode.SEMI_AUTO_PARALLEL?

Reply:
Implemented a new version of zero optimizer parallel in  DATA_PARALLEL refer to DeepSpeed. Not use the MindSpore automatic parallel process.